internal/alternate_state.go

```diff
@@ -0,0 +1,36 @@
+package internal
+
+import (
+	"context"
+
+	"github.com/qlik-oss/enigma-go"
+)
+
+// ListAlternateState will return a lists of all alternate states in an app
+func ListAlternateState(ctx context.Context, doc *enigma.Doc) []string {
+	appLayout, _ := doc.GetAppLayout(ctx)
+	return appLayout.StateNames
+}
+
+// AddAlternateState will add a named alternate state in the app
+func AddAlternateState(ctx context.Context, doc *enigma.Doc, alternateStateName string) {
+	err := doc.AddAlternateState(ctx, alternateStateName)
+	if err != nil {
+		FatalErrorf("could not add state %s: %s ", alternateStateName, err)
+	}
+}
+
+// RemoveAlternateState will remove a named alternate state in the app
+func RemoveAlternateState(ctx context.Context, doc *enigma.Doc, alternateStateName string) {
+	states := ListAlternateState(ctx, doc)
+	stateNameExists := Contains(states, alternateStateName)
+
+	if !stateNameExists {
+		FatalErrorf("no state with the name '%s' found in the app", alternateStateName)
```

Comment:
Maybe add `alternate` to this error message to make it clearer.

Reply:
Fixed